tests/test_core.py

```diff
@@ -8261,12 +8262,14 @@ def test_pthread_create_proxy(self):
   def test_pthread_create_embind_stack_check(self):
     # embind should work with stack overflow checks (see #12356)
     self.set_setting('STACK_OVERFLOW_CHECK', 2)
+    self.set_setting('EXIT_RUNTIME')
     self.emcc_args += ['--bind']
     self.do_run_in_out_file_test('core/pthread/create.cpp')
 
   @node_pthreads
   def test_pthread_exceptions(self):
     self.set_setting('PTHREAD_POOL_SIZE', '2')
+    self.set_setting('EXIT_RUNTIME')
```

Comment:
If we need `EXIT_RUNTIME` in basically all the pthreads tests on node, I wonder if maybe we should just enforce that, that is, error if node pthreads are run without it. There may not be a good enough reason to support pthreads without that.

Reply:
Even if we need this in all our unit test I don't think that necessarily means that every program wants that behaviour.    Basically its always desirable for our unittests to exit the process when they are done, but that isn't true for JS programs in general.

Reply:
Fair enough. I worry this might be confusing but I agree we can't simplify it as easily as I thought.